rasa/core/training/story_reader/yaml_story_reader.py

```diff
@@ -0,0 +1,245 @@
+                    docs=DOCS_URL_STORIES,
+                )
+                continue
+
+            if KEY_PLAIN_STORY_NAME in story_item.keys():
+                self._parse_plain_story(story_item)
+
+    def _parse_plain_story(self, story_item: Dict[Text, Any]) -> None:
+        story_name = story_item.get(KEY_PLAIN_STORY_NAME, "")
+        if not story_name:
+            raise_warning(
+                f"Issue found in {self.source_name}: \n"
+                f"{story_item}\n"
+                f"The story has an empty name."
+                f"Stories should have a name defined under `{KEY_PLAIN_STORY_NAME}` key."
+                "It will be skipped.",
+                docs=DOCS_URL_STORIES,
+            )
+
+        steps = story_item.get(KEY_PLAIN_STORY_STEPS, "")
+
+        self._new_story_part(story_name, self.source_name)
+
+        for step in steps:
+            self._parse_step(step)
+
+    def _parse_step(self, step: Dict[Text, Any]) -> None:
+
+        if KEY_PLAIN_STORY_USER_UTTERANCE in step.keys():
+            self._parse_plain_story_user_utterance(step)
+        elif KEY_OR in step.keys():
+            self._parse_or_statement(step)
+        elif KEY_SLOT_NAME in step.keys():
+            self._parse_slot(step)
+        elif KEY_ACTION in step.keys():
+            self._parse_action(step)
+        elif KEY_CHECKPOINT in step.keys():
+            self._parse_checkpoint(step)
+        elif KEY_METADATA in step.keys():
+            pass
+        else:
+            raise_warning(
+                f"Issue found in {self.source_name}: \n"
+                f"Found an unexpected step in the story description:\n"
+                f"{step}\n"
+                "It will be skipped.",
+                docs=DOCS_URL_STORIES,
+            )
+
+    def _parse_plain_story_user_utterance(self, step: Dict[Text, Any]) -> None:
+
+        if self.use_e2e:
+            # TODO
+            pass
+        else:
+            utterance = self._parse_raw_user_utterance(step)
+            self.current_step_builder.add_user_messages([utterance])
+
+    def _parse_or_statement(self, step):
+        utterances = []
+
+        for sub_step in step.get(KEY_OR):
+            if KEY_PLAIN_STORY_USER_UTTERANCE in sub_step.keys():
+                utterance = self._parse_raw_user_utterance(sub_step)
+                if utterance:
+                    utterances.append(utterance)
+            else:
+                raise_warning(
+                    f"Issue found in {self.source_name}: \n"
+                    f"`OR` statement can only have `{KEY_PLAIN_STORY_USER_UTTERANCE}` "
+                    f"as a sub-element. This step will be skipped:\n"
+                    f"`{sub_step}`\n",
+                    docs=DOCS_URL_STORIES,
+                )
+                return
+
+        self.current_step_builder.add_user_messages(utterances)
+
+    def _parse_raw_user_utterance(self, step: Dict[Text, Any]) -> UserUttered:
+
```

Comment:
optional: I'm fine with the empty lines after method headers, but we should aim for consistency in the code base (I believe that would be dropping them)